service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderController.java

```diff
@@ -142,15 +146,17 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
     }
 
     @PostMapping("/submitted")
-    public void handleSubmittedEvent(@RequestHeader(value = "authorization") String authorization,
-                                     @RequestHeader(value = "user-id") String userId,
-                                     @RequestBody CallbackRequest callbackRequest) {
+    public void handleSubmittedEvent(@RequestBody CallbackRequest callbackRequest) {
         CaseData caseData = mapper.convertValue(callbackRequest.getCaseDetails().getData(), CaseData.class);
         String mostRecentUploadedDocumentUrl = service.getMostRecentUploadedOrderDocumentUrl(
             caseData.getOrderCollection());
 
-        applicationEventPublisher.publishEvent(new GeneratedOrderEvent(callbackRequest, authorization, userId,
-            concatGatewayConfigurationUrlAndMostRecentUploadedOrderDocumentPath(mostRecentUploadedDocumentUrl)));
+        byte[] documentContents = documentDownloadService.downloadDocument(mostRecentUploadedDocumentUrl);
+
+        applicationEventPublisher.publishEvent(
+            new GeneratedOrderEvent(callbackRequest, requestData.authorisation(), requestData.userId(),
+                concatGatewayConfigurationUrlAndMostRecentUploadedOrderDocumentPath(mostRecentUploadedDocumentUrl),
```

Comment:
this method name seems incredibly verbose... I mean does it just concat a string? why is this extracted?

Reply:
Not sure, I think @Gogetter made this one a while back, I'm guessing it's extracted to catch the URISyntaxException, happy to change it and do it within handleSubmittedEvent without extracting

Reply:
@joez-f the method prefixes the gateway url (does not require user to login I think) with the path from `mostRecentUploadedDocumentUrl` (which is doc store and requires login). This given we are sending notification to people with no case access.